sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +190,37 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is not auto {use this}
+        # B. freq is auto
+        #     B1. freq is infered {use this}
+        #     B2. freq is not infered
+        #         B2.1. y is date-like {use "D"}
+        #         B2.2. y is int-like  {use 1}
+        #         B2.3. other indexes; this part of code is never reached
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
+                    # B.2.1
+                    self._freq = "D"
+                elif isinstance(y.index, (pandas.Index, pandas.RangeIndex)):
+                    # B.2.2
+                    self._freq = 1
```

Comment:
I think we should split this case depending on `y.index.diff(periods=1).unique()`.

1. If it has only 2 elements (`pandas.NaT` and some integer), use that integer.
2. Otherwise, raise an error.

Reply:
https://github.com/sktime/sktime/pull/6039#issuecomment-1979301666
> neuralforecast uses freq only to generate dates in prediction horizon and don't use otherwise

even without auto it works with any time-like index, irrelevant to the freq value or missing data. So we are safe to just return 1 here, but if you suggest it to be correct not just be working, then I think we should do this but it will have no effect on the results.

Reply:
I think it'll be safer to be correct in case the `neuralforecast` logic changes in future. For all I know, it may have changed already in `1.7.0`, as I commented the linked text about `1.6.4`.

Did you verify this in the latest release as well?

Reply:
> Did you verify this in the latest release as well?

Yes, I verified that, but this sure makes sense
> I think it'll be safer to be correct in case the neuralforecast logic changes in future